docs/DeveloperGuide.md

```diff
@@ -5,25 +5,115 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Overview
+![Overview.png](Overview.png)
```

Comment:
As per the AB3 DG, perhaps you could consider representing your overall architecture using components instead of individual classes. It might help your diagram look a little neater.